src/pemfile.rs

```diff
@@ -64,10 +83,30 @@ pub fn read_one(rd: &mut dyn io::BufRead) -> Result<Option<Item>, io::Error> {
             Some(line.as_slice())
         };
 
-        if let ControlFlow::Break(opt) = read_one_impl(next_line, &mut section, &mut b64buf)? {
-            return Ok(opt);
-        } else {
-            continue;
+        match read_one_impl(next_line, &mut section, &mut b64buf) {
+            Ok(ControlFlow::Break(opt)) => return Ok(opt),
+            Ok(ControlFlow::Continue(())) => continue,
+            Err(e) => {
+                let e = match e {
```

Comment:
Style nit: if you forego the `e` binding in favor of yielding `Err(match e { ..})` directly this will format a bit more nicely.